src/commandExecutor.ts

```diff
@@ -95,11 +102,31 @@ const CommandExecutor = async () => {
     ))
   }
 
+  const _validateAndParseRawTx = (rawTxCborHex: CborHex): RawTransaction => {
+    const rawTxCbor = Buffer.from(rawTxCborHex, 'hex')
+    const validationErrrors = validateRawTx(rawTxCbor)
+    validationErrrors.forEach((validationError) => {
+      if (!validationError.fixable) {
+        throw Error(validationError.reason)
+      }
+    })
+    if (validationErrrors.length !== 0) {
+      // eslint-disable-next-line no-console
+      console.log('Note: following errors in the transaction will be fixed:')
+      validationErrrors.forEach((validationError) => {
+        // eslint-disable-next-line no-console
+        console.log(`${validationError.reason} (${validationError.position})`)
+      })
+    }
+    return transformRawTx(parseRawTx(rawTxCbor))
+  }
```

Comment:
There are a few issues with this:
1. I was not sure whether we want to show the `Note: ...`. I think we should show it at least with `createTxWitnesses` call - the user should know that he is signing a different CBOR, since it's likely he is going to create non-hw witnesses as well and then assembling them.
2. Unfortunately, the `nativeScriptWitnesses` CBOR field in cardano-cli transaction is an indefinite-length array (even when empty), so the `CBOR_IS_NOT_CANONICAL` fixable error is *always* present (relevant interop-lib code [here](https://github.com/vacuumlabs/cardano-hw-interop-lib/blob/prepare-for-integration/src/txValidators.ts#L200-L204)). I feel like we should solve this in interop-lib (it doesn't make sense to hack around this issue in hw-cli when interop-lib's `validateRawTx` function should take this into account), but I'm not sure how.

@KubqoA @janmazak what do you think?

Reply:
> What does it actually mean to transform raw tx? Isn't it true that we only want to transform the tx body?

This exactly. But I don't know how to modify interop-lib's code to only parse and validate the txBody part of the rawTx. The `cbor` lib has a `decodeFirst` function, but it seems to be doing something different.

Reply:
Resolved :tada: